events/janus_gelfevh.c

```diff
@@ -0,0 +1,476 @@
+		error_code = JANUS_GELFEVH_ERROR_INVALID_REQUEST;
+		g_snprintf(error_cause, 512, "Unknown request '%s'", request_text);
+	}
+
+plugin_response:
+		{
+			json_t *response = json_object();
+			if(error_code == 0) {
+				/* Return a success */
+				json_object_set_new(response, "result", json_integer(200));
+			} else {
+				/* Prepare JSON error event */
+				json_object_set_new(response, "error_code", json_integer(error_code));
+				json_object_set_new(response, "error", json_string(error_cause));
+			}
+			return response;
+		}
+}
+
+/* Thread to handle incoming events */
+static void *janus_gelfevh_handler(void *data) {
+	JANUS_LOG(LOG_VERB, "Joining GelfEventHandler handler thread\n");
+	json_t *event = NULL, *output = NULL;
+	static char *event_text = NULL;
+	const char *short_message = NULL;
+
+	while(g_atomic_int_get(&initialized) && !g_atomic_int_get(&stopping)) {
+		event = g_async_queue_pop(events);
+		if(event == NULL)
+			continue;
+		if(event == &exit_event)
+			break;
+		output = NULL;
+
+		while(TRUE) {
+			/* Handle event */
+			int type = json_integer_value(json_object_get(event, "type"));
+			switch(type) {
+				case JANUS_EVENT_TYPE_SESSION:
+					short_message = "JANUS_EVENT_TYPE_SESSION";
+					break;
+				case JANUS_EVENT_TYPE_HANDLE:
+					short_message = "JANUS_EVENT_TYPE_HANDLE";
+					break;
+				case JANUS_EVENT_TYPE_JSEP:
+					short_message = "JANUS_EVENT_TYPE_JSEP";
+					break;
+				case JANUS_EVENT_TYPE_WEBRTC:
+					short_message = "JANUS_EVENT_TYPE_WEBRTC";
+					break;
+				case JANUS_EVENT_TYPE_MEDIA:
+					short_message = "JANUS_EVENT_TYPE_MEDIA";
+					break;
+				case JANUS_EVENT_TYPE_PLUGIN:
+					short_message = "JANUS_EVENT_TYPE_PLUGIN";
+					break;
+				case JANUS_EVENT_TYPE_TRANSPORT:
+					short_message = "JANUS_EVENT_TYPE_TRANSPORT";
+					break;
+				case JANUS_EVENT_TYPE_CORE:
+				case JANUS_EVENT_TYPE_EXTERNAL:
+					short_message = "JANUS_EVENT_TYPE_CORE";
+					break;
+				default:
+					JANUS_LOG(LOG_WARN, "Unknown type of event '%d'\n", type);
+					short_message = "UNKNOWN_JANUS_EVENT";
+					break;
+			}
+			output = event;
+			/* Add custom fields */
+			json_t *microtimestamp = json_object_get(event, "timestamp");
+			if(microtimestamp && json_is_integer(microtimestamp)) {
+				double created_timestamp = (double)json_integer_value(microtimestamp) / 1000000;
+				json_object_set(output, "timestamp", json_real(created_timestamp));
+			}
+			else {
+				struct timeval t;
+				gettimeofday(&t, NULL);
+				double micro_timestamp = (double)(1000000 * t.tv_sec + t.tv_usec) / 1000000;
+				json_object_set(output, "timestamp", json_real(micro_timestamp));
+			}
+			json_object_set(output, "version", json_string("1.1"));
+			json_object_set(output, "host", json_string("janus"));
+			json_object_set(output, "level", json_integer(1));
+			json_object_set(output, "short_message", json_string(short_message));
+			json_object_set(output, "full_message", json_object_get(event, "event"));
```

Comment:
I believe that when you use `json_object_set`, a reference is already added to that, so you shouldn't have issues when switching to a new object for `output`. Anyway, you may want to make sure that `full_message` is indeed there when you make that change.